src/pages/Signin/Signin.jsx

```diff
@@ -1,44 +1,107 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
+import { Loader } from "../../components/Loader/Loader";
+import { useAuth } from "../../context";
 
 export const Signin = () => {
+  const { signin } = useAuth();
+  const [formData, setFormData] = useState({
+    email: "",
+    password: "",
+    rememberMe: false,
+  });
+  const { email, password, rememberMe } = formData;
+  const isFormFullyFilled = email && password;
+  const formChangeHandler = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const [loading, setLoading] = useState(false);
+
+  const formSubmitHandler = async (e) => {
+    e.preventDefault();
+    try {
+      setLoading(true);
+      if (!isFormFullyFilled) return;
+      await signin(formData);
+      setLoading(false);
+    } catch (err) {
+      setLoading(false);
+      console.error(err);
+    }
+  };
   return (
     <>
       <header className="header">
         <p className="font-bold text-center">Sign In🔐</p>
       </header>
       <div className="signup-card">
-        <div className="input-group">
-          <label htmlFor="username">Username</label>
-          <input
-            type="text"
-            className="input"
-            placeholder="Enter Username:"
-            name="username"
-          />
-        </div>
-        <div className="input-group">
-          <label htmlFor="password">Password</label>
-          <input
-            type="password"
-            className="input"
-            placeholder="Enter Password: "
-            name="password"
-          />
-        </div>
-        <div className="flex justify-between text-xs align-center">
-          <div className="flex align-center">
-            <input type="checkbox" name="remember-me" />
-            <label htmlFor="remember-me">&#160;Remember Me</label>
+        <form action="">
+          <div className="input-group">
+            <label htmlFor="email">Email</label>
+            <input
+              type="text"
+              id="email"
+              className="input"
+              placeholder="Enter Email:"
+              name="email"
+              onChange={formChangeHandler}
+              value={email}
```

Comment:
I think you should use the `required` attribute here to let users know that these values need to fill.